velox/exec/tests/ColumnStatsCollectorTest.cpp

```diff
@@ -0,0 +1,502 @@
+      {std::vector<std::string>{"c0"},
+       std::vector<std::string>{"count(c1)", "sum(c2)"},
+       core::AggregationNode::Step::kPartial,
+       true},
+      // Multiple grouping keys.
+      {std::vector<std::string>{"c0", "c1"},
+       std::vector<std::string>{"sum(c2)"},
+       core::AggregationNode::Step::kSingle,
+       false},
+      {std::vector<std::string>{"c0", "c1"},
+       std::vector<std::string>{"sum(c2)"},
+       core::AggregationNode::Step::kPartial,
+       false},
+      {std::vector<std::string>{"c0", "c1"},
+       std::vector<std::string>{"sum(c2)"},
+       core::AggregationNode::Step::kSingle,
+       true},
+      {std::vector<std::string>{"c0", "c1"},
+       std::vector<std::string>{"sum(c2)"},
+       core::AggregationNode::Step::kPartial,
+       true}};
+
+  for (const auto& testCase : testCases) {
+    SCOPED_TRACE(testCase.debugString());
+
+    // Create stats spec for count and sum
+    auto statsSpec = createStatsSpec(
+        inputType, testCase.groupingKeys, testCase.step, testCase.aggregates);
+
+    // Create collector.
+    auto collector = createStatsCollector(statsSpec, inputType);
+
+    // Create test data.
+    auto inputBatches = createTestData(inputType, 10, 100, testCase.hasNulls);
+    VELOX_ASSERT_THROW(collector->addInput(inputBatches[0]), "");
+
+    // Initialize.
+    collector->initialize();
+
+    // Add input.
+    for (const auto& input : inputBatches) {
+      collector->addInput(input);
+      ASSERT_EQ(collector->getOutput(), nullptr);
+    }
+    ASSERT_FALSE(collector->finished());
+    collector->noMoreInput();
+    ASSERT_FALSE(collector->finished());
+
+    // Get output
+    std::vector<RowVectorPtr> outputBatches;
+    for (;;) {
+      auto output = collector->getOutput();
+      if (output != nullptr) {
+        // Single output row without grouping keys.
+        if (testCase.groupingKeys.empty()) {
+          ASSERT_EQ(output->size(), 1);
+        }
+        ASSERT_EQ(
+            output->childrenSize(),
+            testCase.aggregates.size() + testCase.groupingKeys.size());
+        // Verify column names (using asRowType to access field names)
+        auto rowType = std::dynamic_pointer_cast<const RowType>(output->type());
+        ASSERT_NE(rowType, nullptr);
+        auto outputChannel = testCase.groupingKeys.size();
+        const int numAggregates = testCase.aggregates.size();
+        for (int i = 0; i < numAggregates; ++i) {
+          EXPECT_EQ(
+              rowType->nameOf(outputChannel++), statsSpec.aggregateNames[i]);
+        }
+        outputBatches.push_back(output);
+        EXPECT_FALSE(collector->finished());
+      } else {
+        break;
+      }
+    }
+    // Verify finished.
+    EXPECT_TRUE(collector->finished());
+
+    verifyResult(
+        inputBatches,
+        testCase.groupingKeys,
+        testCase.step,
+        testCase.aggregates,
+        outputBatches);
+  }
+}
+
+TEST_F(ColumnStatsCollectorTest, outputBatchSize) {
```

Comment:
What are we testing here? Why is it important to test output batch size?

Reply:
Removed.